src/utils/word-api.ts

```diff
@@ -17,42 +30,31 @@ class WordApi {
     });
   }
 
-  static async getJabRefCitations(): Promise<Array<Word.ContentControl> | void> {
-    return Word.run(async (context: Word.RequestContext) => {
-      const { contentControls } = context.document;
-      context.load(contentControls, "tag, length");
-      await context.sync();
-      return contentControls.items.filter(
-        (citation) =>
-          citation.tag.includes("JABREF-CITATION") &&
-          !citation.tag.includes("NEW")
-      );
-    }).catch((error) => {
-      console.log(`Error: ${JSON.stringify(error)}`);
-      if (error instanceof OfficeExtension.Error) {
-        console.log(`Debug info: ${JSON.stringify(error.debugInfo)}`);
-      }
-    });
+  static async getJabRefCitations(
+    context: Word.RequestContext
+  ): Promise<Array<Word.ContentControl>> {
+    const { contentControls } = context.document;
+    context.load(contentControls, "tag, length");
+    await context.sync();
+    return contentControls.items.filter((citation) =>
+      citation.tag.includes("JABREF-CITATION")
+    );
   }
 
   static async getPositionOfNewCitation(): Promise<number | void> {
     return Word.run(async (context: Word.RequestContext) => {
-      const { contentControls } = context.document;
-      context.load(contentControls, "tag, length");
+      const currentPosition = context.document.getSelection();
+      const jabRefCitations = await WordApi.getJabRefCitations(context);
+      const locationArray = jabRefCitations.map((citation) => {
+        const citationToCompareWith = citation.getRange("Start");
+        const currentSelectionRange = currentPosition.getRange("Whole");
+        return citationToCompareWith.compareLocationWith(currentSelectionRange);
+      });
```

Comment:
Is the following context sync really needed? Since you don't access the word api afterwards...

Reply:
This location array only store reference to the API result. It doesn't have any results right now. If we try to read value with calling context sync, then it throws an error. The context Sync method sends all queued commands to Word for execution.